### ISSUE TITLE
ClientModel: Migrate stable types from Azure.Core Shared source to public types

### ISSUE THREAD

**Issue by christothes** [Azure/azure-sdk-for-net#28369]
### Library name

Azure.Core

## Types

- [ ] **ClientDiagnostics** (allows removal of [reflection](https://github.com/Azure/azure-sdk-for-net/blob/bb449821bd23d24357c9e4118b373fa9b7801516/sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs#L388) in DiagnosticsScope. 
- [ ] [Files in azure-sdk-for-net/sdk/core/Azure.Core/src/Shared](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core/src/Shared)

Note: shared source in the autorest.csharp repo is now tracked by https://github.com/Azure/azure-sdk-for-net/issues/32006

**Comment by annelo-msft** [comment]
Includes migrating Generator Shared source to Azure.Core public types as well.

Related to:
- https://github.com/Azure/autorest.csharp/issues/2231
- https://github.com/Azure/autorest.csharp/issues/2148

In the Zinc timeframe, we plan to make a plan for this migration.

**Comment by annelo-msft** [comment]
Related to https://github.com/Azure/azure-sdk-for-net/issues/30714

**Comment by heaths** [comment]
As @christothes brought up in triage, perhaps this could be broken up into bite-sized issues - some of which could be done in MQ potentially.

**Comment by annelo-msft** [comment]
Specific customer ask for MultipartFormDataContent to be made public in https://github.com/Azure/azure-sdk-for-net/issues/24819

**Comment by pallavit** [comment]
We have had a few issues where we have considered having Helper\Utility APIs that may be useful to other SDKs and we should consider exposing those APIs are public helper methods in core.

**Comment by annelo-msft** [comment]
> We have had a few issues where we have considered having Helper\Utility APIs that may be useful to other SDKs and we should consider exposing those APIs are public helper methods in core.

@pallavit, would it be possible to link those issues to this one so we can track them together?

**Comment by pallavit** [comment]
Why is this a P1? For instance, why supporting this is higher priority than enabling support for pagination in debranded toolchain?

**Comment by annelo-msft** [comment]
> Why is this a P1? For instance, why supporting this is higher priority than enabling support for pagination in debranded toolchain?

My initial impression is that for our efforts to parallelize Core work, it'd be good to have a plan here to unblock forward progress as we chip away on these types.  For pagination, I don't know of a concrete scenario, but I'm open to repriorization based on new information, so please view what we have today as the start of the conversation and happy to reshuffle as needed!

**Comment by annelo-msft** [comment]
TODO: consider adding a helper to polyfill missing APIs on netstandard2.0 in serialization routines, e.g. to use one type for instances of https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-net+NET6_0_OR_GREATER+NOT+path%3AResourceManager&type=code&p=3

**Comment by annelo-msft** [comment]
Issue still active and slated for this semester

**Comment by jsquire** [comment]
> Issue still active and slated for this semester

After 30 days of no activity, the bot will close this again.   I'm going to clone the issue in order to buy us two more years of peace.

**Comment by jsquire** [comment]
New issue: # #43760
